microcosm_pubsub/tests/chain/test_decorators.py

```diff
@@ -0,0 +1,147 @@
+from hamcrest import (
+    assert_that,
+    calling,
+    equal_to,
+    is_,
+    raises,
+)
+from marshmallow import ValidationError
+
+from microcosm_pubsub.chain.decorators import binds, extracts
+from microcosm_pubsub.chain.context_decorators import (
+    get_from_context,
+    save_to_context,
+    save_to_context_by_func_name,
+    temporary_replace_context_keys,
+)
+
+
+class TestDecorators:
+
+    def test_get_from_context(self):
+        context = dict(arg=123)
+        wrapped = get_from_context(context, lambda arg: arg)
+        assert_that(wrapped(), is_(123))
+
+    def test_save_to_context(self):
+        context = dict()
+
+        @extracts("arg")
+        def extractarg(number):
+            return number
+
+        @extracts("arg1", "arg2")
+        def extractarg1_and_arg2(num1, num2):
+            return num1, num2
+
+        @extracts("args")
+        def extractargs(num1, num2):
+            return num1, num2
+
+        save_to_context(context, extractarg)(0)
```

Comment:
What happens if `save_to_context*` are passed something that's not decorated? These feel like sunny day cases only.

Reply:
Will add a test